projects/plugins/protect/src/js/routes/home/index.jsx

```diff
@@ -13,16 +15,25 @@ import styles from './styles.module.scss';
  */
 const HomePage = () => {
 	const { data: status } = useScanStatusQuery( { usePolling: true } );
-	const { core, plugins, themes, files } = status;
 
-	const data = [
-		core,
-		...plugins,
-		...themes,
-		{ checked: true, threats: files, type: 'files' },
-	].map( ( item, index ) => {
-		return { id: index + 1, ...item };
-	} );
+	const data = useMemo(
+		() => [
+			...( Object.keys( status.core ).length ? [ status.core ] : [] ),
+			...status.plugins,
+			...status.themes,
+			...( status.dataSource === 'scan_api'
+				? [
+						{
+							checked: !! status.lastChecked,
+							threats: status.files,
+							type: 'files',
+							name: __( 'Files', 'jetpack-protect' ),
```

Comment:
As I described [here](https://github.com/Automattic/jetpack/pull/40602#discussion_r1887560400), I think this might not be necessary. The issue adding this resolved was the Files entry in list view was unclear as to what it referred to, adding this conditional for that single state was ideal, but adding it more globally introduces some redundancies (eg the table view include `Type` and `Name` by default, so we'll have Files twice in that row). I propose we just include `Type` by default as list view field and make `enableHiding: false`

Reply:
Ensuring the `Type` field is not hidden in list view sounds good enough to me :+1:

Communicating the limited information we have on hand about the scanned files is definitely a little tricky in the context of a dataview item.

Reply:
```suggestion
```